src/Harald.WebApi/Features/Connections/Infrastructure/DrivingAdapters/Api/ConnectionsController.cs

```diff
@@ -70,11 +82,102 @@ public class ConnectionsController : ControllerBase
                     new {message = validationException.MessageToUser}
                 );
             }
-
-
+            
             var connectionDtos = connections.Select(ConnectionDto.CreateFromConnection);
 
             return Ok(new ItemsEnvelope<ConnectionDto>(connectionDtos));
         }
+
+        [HttpPost]
+        public async Task<IActionResult> AddConnection(ConnectionDto connection)
+        {
+            if (string.IsNullOrEmpty(connection.ClientId))
```

Comment:
We could do these validations on the object factories and return 422 instead of the 400

Reply:
Personally I always keep anything related to the protocol (in this case HTTP) in the controller, as it has the responsibility of mapping input to/from the protocol and the status code is very much part of this in my world. Also if we start putting HttpStatusCode objects inside our core factory classes their dependencies will end up all over the place. So I would prefere leaving it as is and sticking to throwing exceptions or returning null from our factories.